ValueError: invalid literal for int() with base 10: 'fifo:/tmp/GMfifo519308'

When I run pytest-jobserver I get an error
```
$ make -j3 test-python-pytest
[...]
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/_pytest/main.py", line 269, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1074, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pluggy/_hooks.py", line 523, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self._hookimpls.copy(), kwargs, False)
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pluggy/_manager.py", line 119, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pluggy/_callers.py", line 138, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pluggy/_callers.py", line 102, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pytest_jobserver/plugin.py", line 58, in pytest_configure
INTERNALERROR>     or jobserver_from_env_make(config)
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pytest_jobserver/configure.py", line 65, in jobserver_from_env_make
INTERNALERROR>     fd_read, fd_write = tuple(FileDescriptor(int(fd)) for fd in fds.split(","))
INTERNALERROR>   File "/home/vbraun/Talque-4/tools/talque3/lib/python3.10/site-packages/pytest_jobserver/configure.py", line 65, in <genexpr>
INTERNALERROR>     fd_read, fd_write = tuple(FileDescriptor(int(fd)) for fd in fds.split(","))
INTERNALERROR> ValueError: invalid literal for int() with base 10: 'fifo:/tmp/GMfifo519308'
make: *** [Makefile:295: test-python-pytest] Error 3
```
Presumably the `MAKEFLAGS` format changed (actual value is `-j3 --jobserver-auth=fifo:/tmp/GMfifo519308`). For the record, my make version is
```
$ make --version
GNU Make 4.4.1
Built for x86_64-redhat-linux-gnu
Copyright (C) 1988-2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```
There is now a library to just interact with the Make jobserver at https://github.com/milahu/gnumake-tokenpool, one idea might be to use that instead of rolling your own `MAKEFLAGS` parser.

--- a/pytest_jobserver/configure.py
+++ b/pytest_jobserver/configure.py
@@ -53,13 +53,16 @@ def jobserver_from_env_make(config: Config) -> Optional[FileDescriptorsRW]:
     parser.add_argument("--jobserver-auth", default=None)
     args, _ = parser.parse_known_args(shlex.split(makeflags))
 
+    if args.jobserver_auth is not None and args.jobserver_auth.startswith("fifo:"):
+        return path_to_file_descriptors(args.jobserver_auth[5:])
+
     fds = args.jobserver_fds or args.jobserver_auth
     if fds is None:
         return None
 
     if config.pluginmanager.hasplugin("xdist"):
         raise pytest.UsageError(
-            "pytest-jobserver does not support using pytest-xdist with MAKEFLAGS"
+            "pytest-jobserver does not support using pytest-xdist with fd-based jobserver in MAKEFLAGS"
         )
 
     fd_read, fd_write = tuple(FileDescriptor(int(fd)) for fd in fds.split(","))

--- a/pytest_jobserver/test/test_plugin.py
+++ b/pytest_jobserver/test/test_plugin.py
@@ -62,6 +62,23 @@ def test_config_env_pytest(testdir: TestDir) -> None:
     assert result.ret == 0
 
 
+def test_config_makeflags_pytest(testdir: TestDir) -> None:
+    testdir.makepyfile("""
+        def test_pass(request):
+            pass
+    """)
+    make_jobserver(testdir.tmpdir, "jobserver_fifo", 1)
+    testdir.monkeypatch.setenv("MAKEFLAGS", "--jobserver-auth=fifo:jobserver_fifo")
+
+    result = testdir.runpytest("-v")
+
+    result.stdout.fnmatch_lines(["*::test_pass PASSED*"])
+    result.stdout.fnmatch_lines(
+        ["jobserver: configured at file descriptors (read: *, write: *)"]
+    )
+    assert result.ret == 0
+
+
 def test_jobserver_token_fixture(testdir: TestDir) -> None:
     testdir.makepyfile("""
         def test_value(jobserver_token: int):
@@ -84,7 +101,9 @@ def test_xdist_makeflags_fails(testdir: TestDir) -> None:
     result = testdir.runpytest("-v", "-n2")
     assert result.ret == 4, "Expected pytest would fail to run with MAKEFLAGS and xdist"
     result.stderr.fnmatch_lines(
-        ["ERROR: pytest-jobserver does not support using pytest-xdist with MAKEFLAGS"]
+        [
+            "ERROR: pytest-jobserver does not support using pytest-xdist with fd-based jobserver in MAKEFLAGS"
+        ]
     )
 
 

